test/src/unit-buffer.cc

```diff
@@ -187,3 +187,63 @@ TEST_CASE("Buffer: Test move", "[buffer]") {
   CHECK(b.alloced_size() == 0);
   CHECK(b.data() == nullptr);
 }
+
+TEST_CASE("ConstBuffer: Overflow on read", "[buffer][ConstBuffer]") {
+  char data[3] = {1, 2, 3};
+  auto buff = ConstBuffer(data, 3);
+
+  CHECK(buff.size() == 3);
+  buff.advance_offset(2);
+  uint64_t n = std::numeric_limits<uint64_t>::max() - 1;
+  auto st = buff.read(data, n);
+  REQUIRE(!st.ok());
+}
+
+TEST_CASE("Buffer: Overflow on read", "[buffer][Buffer]") {
+  char data[3] = {1, 2, 3};
+  auto buff = Buffer(data, 3);
+
+  CHECK(buff.size() == 3);
+  buff.advance_offset(2);
+  uint64_t n = std::numeric_limits<uint64_t>::max() - 1;
+  auto st = buff.read(data, n);
+  REQUIRE(!st.ok());
+}
+
+// Disabled by default with [.]
```

Comment:
> overlooking something

I would have overlooked it too. The implementation uses `assert()`. On windows that raises a non-C++ exception (SEH), which aborts the program, so that other unit tests don't run. In other words `[!shouldfail]` doesn't work here, because Catch would never see the an exception in its failure handler.

I thought about writing a replacement for `assert()` that would throw in a unit test, but I decided that that's beyond the scope of this PR. It's problematic to implement, too, since `assert` is a macro.